webapp/src/main/webapp/src/app/app.component.ts

```diff
@@ -21,26 +22,30 @@ export class AppComponent {
   /*
   Even though the angulartics2GoogleAnalytics variable is not explicitly used, without it analytics data is not sent to the service
    */
-  constructor(public translate: TranslateService, private _userService: UserService,
+  constructor(public translate: TranslateService,
+              private _userService: UserService,
+              private router: Router,
               private angulartics2GoogleAnalytics: Angulartics2GoogleAnalytics) {
 
-    let languages = [ 'en', 'ja' ];
-    let defaultLanguage = languages[ 0 ];
-    translate.addLangs(languages);
+    let defaultLanguage = SupportedLanguages[ 0 ];
+    translate.addLangs(SupportedLanguages);
     translate.setDefaultLang(defaultLanguage);
-    translate.use(languages.indexOf(translate.getBrowserLang()) != -1 ? translate.getBrowserLang() : defaultLanguage);
+    translate.use(SupportedLanguages.indexOf(translate.getBrowserLang()) != -1 ? translate.getBrowserLang() : defaultLanguage);
 
   }
 
   ngOnInit() {
     this._userService.getCurrentUser().subscribe(user => {
 
       this._user = user;
+      let userHasAccess = user.permissions.some(permission => RequiredPermissions.indexOf(permission) !== -1);
```

Comment:
The name required permissions make me think it would have an AND relationship instead of the OR impl here.
Might make sense to make a utility similar to spring's hasAnyAuthority() method so we can make quick sense of what is going on here.
```has(user.permissions, "GROUP_WRITE")``` or ```hasAnyOf(user.permissions, ["GROUP_WRITE"])```

Reply:
I can rename it just to `AdminPermissions` then.  My intent is to self document in one spot what our supported languages are and what are the permissions will be needed.

Reply:
K. We may want to introduce an ```application-settings.ts``` similar to our spring ```applicaiton.yml```
Thoughts?